src/main/java/sirius/web/templates/ClosedChannelHelper.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.web.templates;
+
+import com.lowagie.text.ExceptionConverter;
+import org.asynchttpclient.exception.ChannelClosedException;
+
+import java.nio.channels.ClosedChannelException;
+
+/**
+ * Helps to detect closed channel exceptions in complex exception chains.
+ */
+public class ClosedChannelHelper {
+
+    private ClosedChannelHelper() {
+        // Prevent instantiation
+    }
+
+    /**
+     * Detects if the given exception or any of its causes is a closed channel exception.
+     * <p>
+     * This is useful for handling cases where a channel was closed unexpectedly, such as when a client
+     * disconnects or a network error occurs.
+     * <p>
+     * It also handles the async http client variant and the openpdf exception wrapping.
+     *
+     * @param throwable the exception to check
+     * @return <tt>true</tt> if the exception or any of its causes is a closed channel exception.
+     */
+    public static boolean tryDetectClosedChannelException(Throwable throwable) {
```

Comment:
Kann die Methode so wie sie aktuell implementiert ist nicht zu Endlosschleifen bei zyklischen Exception-Ketten führen?

Reply:
Ja - zyklische Exception-Ketten sind unüblich, aber technisch möglich, wenn man sie manuell oder durch unsauberen Code in Libraries erzeugt.

Ich meine damit im Prinzip sowas:
```
Exception a = new Exception("A");
Exception b = new Exception("B", a);

a.initCause(b); // ZYKLUS: a → b → a
```

Das könnte dann zu einer Endlosrekursion / `StackOverflowError` führen, weil `a` auf `b`, und `b` wieder auf `a` zeigt – das ist ein Zyklus.

Reply:
Ja okay, wenn ich sowas baue, ist aber auch schon ganz schön viel im Argen :)

Ich werd den Code umbauen, es mir aber als Bsp für unser Meeting für Freitag notieren. Das was hier genannt wird sind ja schon eher sehr theoretische Szenarien.